SRE-25 → retire $wgCacheBuster in favour of $wgStyleVersion

mixoteric opened the pull request:
https://wikia-inc.atlassian.net/browse/SRE-25

Per https://www.mediawiki.org/wiki/Manual:$wgStyleVersion _Bump this number when changing the global style sheets and JavaScript. It should be appended in the query string of static CSS and JS includes, to ensure that client-side caches don't keep obsolete copies of global styles._

`$wgCacheBuster` has exactly the same purpose. Let's retire it in favour of core MediaWiki's `$wgStyleVersion`.

This is the first part, that can be merged once the updated version of Wikia/deploytools-plugins is deployed.  The new version will put $wgStyleVersion to `wgCacheBuster.php`.  The file itself will be renamed later. This first part is solely to break circular dependency in naming.

Corresponding config/app changes:
* https://github.com/Wikia/config/pull/2321
* https://github.com/Wikia/config/pull/2322

Corresponding deploytools update:
* to be merged before this change: https://github.com/Wikia/deploytools-plugins/pull/231 :heavy_check_mark: 

Corresponding chef-repo update:
* to be merged before this change: https://github.com/Wikia/chef-repo/pull/15246

In general, I only changed references to variables.  I have not changed public APIs that use various forms of _cache buster_ for naming keys, etc.  There is no practical need to deprecate public APIs, because our internal variable name (not value) changed.

**DO NOT MERGE**

# List of reviewers based on `git log`

* comments changes
    * [e/w/AssetsManager](https://github.com/Wikia/app/pull/14245/commits/776ff2bd602abab194364ae9d4946943c316c5f1) - only comments changed
    * [i/w/Wikia.php](https://github.com/Wikia/app/pull/14245/commits/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5) - only comments changed
* @macbre 
    * [e/ApiExplorer](https://github.com/Wikia/app/commit/60fc90a007b88c64ff6c006152e0a96a2488f325)
    * [e/w/ApiDocs](https://github.com/Wikia/app/commit/e85c37fb6e54b037837472a7da662cfdce9b4642)
    * [e/w/ImageServing](https://github.com/Wikia/app/pull/14245/commits/dc070b383807ec24bb2a11104b4d925a8b145ca8)
* @TK-999 
    * [e/ApiExplorer](https://github.com/Wikia/app/commit/60fc90a007b88c64ff6c006152e0a96a2488f325)
    * [e/w/ApiDocs](https://github.com/Wikia/app/commit/e85c37fb6e54b037837472a7da662cfdce9b4642)
    * [i/w/ui](https://github.com/Wikia/app/pull/14245/commits/76522487cca88d878ce3ba2a4d683917d9c904fc)
* @rogatty 
    * [e/w/ApiDocs](https://github.com/Wikia/app/commit/e85c37fb6e54b037837472a7da662cfdce9b4642)
* @hakubo 
    * [e/w/MercuryApi](https://github.com/Wikia/app/pull/14245/commits/d47cc8640366bcee23acb6a662e33cb1888a8f0a)
    * [e/w/WikiaMobile](https://github.com/Wikia/app/pull/14245/commits/96924da72dc447ced3d556437a6f4d0ab95699d0)
* @rybmat 
    * [e/w/MercuryApi](https://github.com/Wikia/app/pull/14245/commits/d47cc8640366bcee23acb6a662e33cb1888a8f0a)

Commit:
SRE-25 → e/ApiExplorer

Commit:
SRE-25 → e/w/ApiDocs

This code uses the term “cache buster” with and without connotation
to $wgCacheBuser.  I only changed bits, where we specifically refer
to the variable.

WikiaTech commented on extensions/ApiExplorer/SpecialApiExplorer.php:
```diff
@@ -50,7 +50,7 @@ public function execute( $par = null ) {
 		wfProfileIn( __METHOD__ );
 
 		// TODO: Make this work for ResourceLoader (Wikia isn't using RL yet at the time of this writing).
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[project-id]: wikia:app

WikiaTech commented:
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MINOR][MINOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L15): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)
1. ![INFO][INFO] [SpecialApiExplorer.php#L9](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L9): Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. ![INFO][INFO] [SpecialApiExplorer.php#L103](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L103): Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [SpecialApiExplorer.php#L104](https://github.com/Wikia/app/blob/e85c37fb6e54b037837472a7da662cfdce9b4642/extensions/ApiExplorer/SpecialApiExplorer.php#L104): Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

WikiaTech commented on extensions/wikia/ApiDocs/services/CachingApiDocsService.php:
```diff
@@ -20,7 +20,7 @@ class CachingApiDocsService implements IApiDocsService {
 	private $cacheTime = 43200; // 60 * 60 * 12
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

Commit:
SRE-25 → e/w/AssetsManager

This is just a change in a header comment to the method I wrote back in 2014
and deliberately used $wgStyleVersion instead of $wgCacheBuster. :-)

WikiaTech commented:
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MINOR][MINOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/ApiExplorer/SpecialApiExplorer.php#L15): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)
1. ![MINOR][MINOR] [AssetsManagerController.class.php#L183](https://github.com/Wikia/app/blob/776ff2bd602abab194364ae9d4946943c316c5f1/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L183): Immediately return this expression instead of assigning it to the temporary variable "$builder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SRE-25 → e/w/ImageServing

WikiaTech commented on extensions/wikia/ImageServing/api/WikiaApiImageServing.php:
```diff
@@ -23,7 +23,7 @@ public function __construct( $main, $action ) {
 	 */
 	public function execute() {
-		global $wgRequest, $wgCacheBuster;
+		global $wgRequest, $wgStyleVersion;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$wgRequest" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app

WikiaTech commented:
SonarQube analysis reported 21 issues
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MINOR][MINOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/dc070b383807ec24bb2a11104b4d925a8b145ca8/extensions/ApiExplorer/SpecialApiExplorer.php#L15): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SRE-25 → e/w/MercuryApi

Note, that I have only replaced references to $wgCacheBuster variable.
I have left cacheBuster key name unchanged as it is a part of a public
API and there is no practical reason to deprecate it.

WikiaTech commented:
SonarQube analysis reported 29 issues
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L5](https://github.com/Wikia/app/blob/d47cc8640366bcee23acb6a662e33cb1888a8f0a/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L5): Class "MercuryApi" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SRE-25 → e/w/WikiaMobile

WikiaTech commented:
SonarQube analysis reported 29 issues
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L5](https://github.com/Wikia/app/blob/96924da72dc447ced3d556437a6f4d0ab95699d0/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L5): Class "MercuryApi" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SRE-25 → i/w/ui

WikiaTech commented:
SonarQube analysis reported 36 issues
* ![MAJOR][MAJOR] 19 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 13 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L5](https://github.com/Wikia/app/blob/76522487cca88d878ce3ba2a4d683917d9c904fc/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L5): Class "MercuryApi" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SRE-25 → i/w/Wikia.php

WikiaTech commented:
SonarQube analysis reported 117 issues
* ![MAJOR][MAJOR] 83 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 26 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L5](https://github.com/Wikia/app/blob/fcef4ee14dcd9ce4f003aa36690f49e8d377b8e5/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L5): Class "MercuryApi" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre commented on extensions/ApiExplorer/SpecialApiExplorer.php:
```diff
@@ -50,7 +50,7 @@ public function execute( $par = null ) {
+		// Wikia interpolates wgStyleVersion in the wgExtensionPath (we rewrite that in varnish because many proxies won't cache things that have "?" in the URL), but other MediaWikis need the style-version in the querystring.
 		$cbSuffix = ( isset( $wgCityId ) ? "?{$wgStyleVersion}" : "" );
 		$wgOut->addScript( "<script type=\"text/javascript\" src=\"{$wgExtensionsPath}/wikia/JavascriptAPI/Mediawiki.js{$cbSuffix}\"></script>" );
 		$wgOut->addScript( "<script type=\"text/javascript\" src=\"{$wgExtensionsPath}/ApiExplorer/apiExplorer.js{$cbSuffix}\"></script>" );
```
```
> echo "<script type=\"text/javascript\" src=\"{$wgExtensionsPath}/ApiExplorer/apiExplorer.js{$cbSuffix}\"></script>"
<script type="text/javascript" src="https://slot1-images.wikia.nocookie.net/__cb123/common/extensions/ApiExplorer/apiExplorer.js"></script>
```

Let's get rid of `$cbSuffix` variable.

mixoteric replied:
:white_check_mark: , see f9505c0fe1701510c7d75b0b6b934b9559a086c7

mszabo-wikia reviewed (changes requested):
One note left

Commit:
SRE-25 → replace Nirvana call with global reference

WikiaTech commented:
SonarQube analysis reported 117 issues
* ![MAJOR][MAJOR] 83 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 26 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L61](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/ApiExplorer/SpecialApiExplorer.php#L61): Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L62](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/ApiExplorer/SpecialApiExplorer.php#L62): Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L79](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/ApiExplorer/SpecialApiExplorer.php#L79): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L81](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/ApiExplorer/SpecialApiExplorer.php#L81): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L5](https://github.com/Wikia/app/blob/1e1ec8e150802717ec0d4f00eb8bc347964cef10/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L5): Class "MercuryApi" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

ghost commented:
@TK-999 — I've replaced Nirvana's `F::app()` with a simple global reference.

Commit:
SRE-25 → address code review

WikiaTech commented on extensions/ApiExplorer/SpecialApiExplorer.php:
```diff
@@ -50,16 +50,15 @@ public function execute( $par = null ) {
 			$buttonHeight = 15;
-			$collapseSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$cbSuffix";
-			$expandSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$cbSuffix";
+			$collapseSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$wgStyleVersion";
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$collapseSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app

WikiaTech commented on extensions/ApiExplorer/SpecialApiExplorer.php:
```diff
@@ -50,16 +50,15 @@ public function execute( $par = null ) {
-			$collapseSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$cbSuffix";
-			$expandSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$cbSuffix";
+			$collapseSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$wgStyleVersion";
+			$expandSrc = "$wgExtensionsPath/ApiExplorer/collapse.png$wgStyleVersion";
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$expandSrc" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

[project-id]: wikia:app

WikiaTech commented:
SonarQube analysis reported 118 issues
* ![MAJOR][MAJOR] 84 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 26 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L15](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/ApiExplorer/SpecialApiExplorer.php#L15): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L49](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/ApiExplorer/SpecialApiExplorer.php#L49): Remove this unused "$wgCityId" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L78](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/ApiExplorer/SpecialApiExplorer.php#L78): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialApiExplorer.php#L80](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/ApiExplorer/SpecialApiExplorer.php#L80): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L43](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L43): Remove this unused "$mustache" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L84](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L84): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [AssetsManagerController.class.php#L138](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/wikia/AssetsManager/AssetsManagerController.class.php#L138): Remove the use of this variable variable "$$templateLanguage". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1599)
1. ![MAJOR][MAJOR] [WikiaApiImageServing.php#L52](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/wikia/ImageServing/api/WikiaApiImageServing.php#L52): Remove this unused "$key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L5](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L5): Class "MercuryApi" has 23 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [MercuryApi.class.php#L53](https://github.com/Wikia/app/blob/f9505c0fe1701510c7d75b0b6b934b9559a086c7/extensions/wikia/MercuryApi/models/MercuryApi.class.php#L53): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
